ros2_serial_example/config/px4_serial_to_ros2_bridge_params.yaml

```
@@ -0,0 +1,166 @@
+ros2_to_serial_bridge:
+    ros__parameters:
+        backend_comms: udp             # Which backend to use for communications;
+                                       # 'uart' and 'udp' are currently supported.
+        device: /dev/pts/6             # Which device to use for communications;
+                                       # only used when backend is 'uart'.
+        baudrate: 0                    # The baudrate to set on the device; if
+                                       # set to 0, setting baudrate is skipped.
+                                       # Only used when backend is 'uart'.
+        udp_recv_port: 2020            # Which UDP port to use for receiving
+                                       # data; must be 1-65535.  Only used when
+                                       # backend is 'udp'.
+        udp_send_port: 2019            # Which UDP port to use for sending
+                                       # data; must be 1-65535.  Only used when
+                                       # backend is 'udp'.
+        read_poll_ms: 250              # How many milliseconds to wait for new
+                                       # data to come in from the serial port.
+                                       # Larger numbers can use less CPU time,
+                                       # but affect the responsiveness of the
+                                       # program.  The default value of 100 is
+                                       # a good compromise between
+                                       # responsiveness and CPU time.
+        ring_buffer_size: 8192         # The number of bytes to use for the
+                                       # internal read buffer.  Larger values
+                                       # allow larger messages to be transmitted
+                                       # at the cost of more memory.
+        write_sleep_ms: 4              # How many milliseconds to sleep in
+                                       # between servicing ROS 2 callbacks.
+                                       # Larger numbers will result in less CPU
+                                       # usage but also some latency in
+                                       # delivering data from ROS 2 to the
+                                       # serial port.  The default of 4 ms
+                                       # (250Hz) is a good compromise between
+                                       # CPU time and latency.  It is not
+                                       # recommended to set this above 100,
+                                       # since this will cause the application
+                                       # to feel sluggish.
+        dynamic_serial_mapping_ms: 0 # How many milliseconds to wait to get
```

```suggestion
        dynamic_serial_mapping_ms: 0  # How many milliseconds to wait to get
```

More importantly, this is currently set to 0, meaning that we'll be using dynamic mapping (and not static mapping), but there are a whole bunch of static topic maps setup below.  That's fine for runtime (as they'll be ignored), but as an example I think it is a bit confusing.  Do you mean to use dynamic or static mapping in this example?

---

Nope, according with comments
 
> If less than 0, dynamic mapping is disabled static mapping and the topics below will be used

I think it's well configure for static mapping

---

> > If less than 0, dynamic mapping is disabled static mapping and the topics below will be used
> 
> I think it's well configure for static mapping

So this is the point that I'm confused about.  The comment says "if *less* than zero", but this configuration is currently setting it to exactly 0, which means wait forever for a dynamic map.

---

This line of code is the key:
https://github.com/osrf/ros2_serial_example/blob/master/ros2_serial_example/src/ros2_to_serial_bridge.cpp#L125

less or equal to zero, right?

---

Oh, I see.  OK, that is actually a bug in the code; that line of code really should be:

```
if (dynamic_serial_mapping_ms >= 0)
```

My suggestion is to change this PR to set `dynamic_mapping_ms` to -1 (which will work for both the current code and any new code), and then open a separate PR to fix that bug.

---

Done ;) and created the PR https://github.com/osrf/ros2_serial_example/pull/74